src/services/completions.ts

```diff
@@ -996,18 +999,26 @@ namespace ts.Completions {
             completionKind = CompletionKind.PropertyAccess;
 
             // Since this is qualified name check its a type node location
-            const isTypeLocation = insideJsDocTagTypeExpression || isPartOfTypeNode(node.parent);
+            const isImportType = isLiteralImportTypeNode(node);
+            const isTypeLocation = insideJsDocTagTypeExpression || (isImportType && !(node as ImportTypeNode).isTypeOf) || isPartOfTypeNode(node.parent);
             const isRhsOfImportDeclaration = isInRightSideOfInternalImportEqualsDeclaration(node);
             const allowTypeOrValue = isRhsOfImportDeclaration || (!isTypeLocation && isPossiblyTypeArgumentPosition(contextToken, sourceFile));
-            if (isEntityName(node)) {
+            if (isEntityName(node) || isImportType) {
                 let symbol = typeChecker.getSymbolAtLocation(node);
                 if (symbol) {
                     symbol = skipAlias(symbol, typeChecker);
 
                     if (symbol.flags & (SymbolFlags.Module | SymbolFlags.Enum)) {
                         // Extract module or enum members
                         const exportedSymbols = Debug.assertEachDefined(typeChecker.getExportsOfModule(symbol), "getExportsOfModule() should all be defined");
-                        const isValidValueAccess = (symbol: Symbol) => typeChecker.isValidPropertyAccess(<PropertyAccessExpression>(node.parent), symbol.name);
+                        const isValidValueAccess = (symbol: Symbol) => {
+                            if (!isImportType) {
+                                return typeChecker.isValidPropertyAccess(<PropertyAccessExpression>(node.parent), symbol.name);
+                            }
+                            else {
+                                return !(node as ImportTypeNode).isTypeOf && !!(symbol.flags & SymbolFlags.Value);

```

Comment:
Dont you still need ```typeChecker.isValidPropertyAccess``` especially ```checkPropertyAccessibility``` needs to be done to avoid not showing non exported stuff?

Reply:
Non-exported symbols aren't returned by `getExportsOfModule` - except for class statics! I'll add a test.